src/expr/src/linear.rs

```diff
@@ -1374,68 +1387,70 @@ pub mod plan {
+                for u in self.upper_bounds.iter() {
+                    match u.eval(datums, &arena) {
+                        Err(e) => {
+                            return Some(Err((e, time, diff)))
+                                .into_iter()
+                                .chain(None.into_iter());
+                        }
+                        Ok(Datum::Numeric(mut d)) => {
+                            // Upper bound must be at least lower bound, whose
+                            // minimum value is `time`. Negative values will
+                            // always be less than `time`, so will never be
+                            // chosen.
+                            if d.0.is_negative() {
+                                continue;
+                            }
+                            let mut cx = numeric::cx_datum();
+                            cx.rescale(&mut d.0, &Numeric::zero());
+
+                            match cx.try_into_u64(d.0) {
+                                Ok(v) => {
+                                    // Replace `upper_bound` if it is none or we
+                                    // find a smaller value.
+                                    if upper_bound_u64.is_none() || upper_bound_u64 > Some(v) {
+                                        upper_bound_u64 = Some(v);
+                                    }
+                                }
+                                // If conversion fails, value is greater than
+                                // u64::MAX, so upper bound should remain None,
+                                // unless we find another, smaller upper bound.
+                                Err(_) => {}
+                            }
+                        }
+                        Ok(Datum::Null) => {
+                            null_eval = true;
+                        }
+                        x => {
+                            panic!("Non-decimal value in temporal predicate: {:?}", x);
                         }
-                    }
-                    Ok(Datum::Null) => {
-                        null_eval = true;
-                    }
-                    x => {
-                        panic!("Non-decimal value in temporal predicate: {:?}", x);
                     }
                 }
             }
 
-            // Force the upper bound to be at least the lower bound.
-            // This should have the effect downstream of making the two equal,
-            // which will result in no output.
-            // Doing it this way spares us some awkward option comparison logic.
-            // This also ensures that `upper_bound_u128` will be at least `time`,
-            // which means "non-negative" / not needing to be clamped from below.
-            if let Some(u) = upper_bound_i128.as_mut() {
-                if *u < lower_bound_i128 {
-                    *u = lower_bound_i128;
-                }
-            }
```

Comment:
I don't think it is appropriate to delete this.

Reply:
Was just waiting to make sure tests passed before pinging you, but this just got bumped down [here](https://github.com/MaterializeInc/materialize/pull/7571/files#diff-3357e6f69bce6cf8b97aba981c281defe94cf8dfd06989a29b1b46f49a71d5c1R1447-R1453)